lib/karafka/core/monitoring/notifications.rb

```diff
@@ -39,9 +39,13 @@ def register_event(event_id)
         end
 
         # Clears all the subscribed listeners
-        def clear
+        def clear(event=nil)
           @mutex.synchronize do
-            @listeners.each_value(&:clear)
+            if event
+              @listeners[event].clear
```

Comment:
The event name is remapped here:
https://github.com/karafka/karafka-core/blob/master/lib/karafka/core/monitoring/notifications.rb#L36

can we extract this and use the same logic so it is consistent in creation / clearing?

Reply:
Also can we also clear `@events_methods_map`?

Reply:
@dorner one more thing. Can we raise an error (define in karafka/core/errors.rb similar to how it is in karafka) so we do not clear a non-existent event? otherwise ppl may make mistakes of clearing events that were not registered in the first place

Reply:
A little confused here... Calling `listeners[event_id].clear` does not clear the _registration_ of the event itself, only the subscribed listeners. So I didn't think we should be doing anything with the `@events_methods_map`.

I've added the error raising on nonexistent key.

Reply:
@dorner you are right. Sorry :facepalm: